addons/GearSwap/libs/closetCleaner.lua

```diff
@@ -0,0 +1,379 @@
+--Copyright (c) 2016~2017, Brimstone
+--All rights reserved.
+
+--Redistribution and use in source and binary forms, with or without
+--modification, are permitted provided that the following conditions are met:
+
+--    * Redistributions of source code must retain the above copyright
+--      notice, this list of conditions and the following disclaimer.
+--    * Redistributions in binary form must reproduce the above copyright
+--      notice, this list of conditions and the following disclaimer in the
+--      documentation and/or other materials provided with the distribution.
+--    * Neither the name of <addon name> nor the
+--      names of its contributors may be used to endorse or promote products
+--      derived from this software without specific prior written permission.
+
+--THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+--ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+--WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+--DISCLAIMED. IN NO EVENT SHALL <your name> BE LIABLE FOR ANY
+--DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+--(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+--LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+--ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+--(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+--SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-- _addon.name = 'closetCleaner'
+-- _addon.version = '0.7'
+-- _addon.author = 'Brimstone'
+-- _addon.commands = {'cc','closetCleaner'}
+
+local cc = {}
+cc.sandbox = {}
+cc.sandbox.windower = setmetatable({}, {__index = windower})
+cc.sandbox.windower.register_event = functions.empty()
+cc.sandbox.windower.raw_register_event = functions.empty()
+cc.sandbox.windower.register_unhandled_command = functions.empty()
+
+register_unhandled_command(function(...)
+    local cmds = {...}
+    for _,v in ipairs(cmds) do
+        if S{'closetCleaner','cc'}:contains(v:lower()) then
+			setmetatable(cc.sandbox, {__index = gearswap.user_env})
+			cc.sandbox.itemsBylongName = T{}
+			cc.sandbox.itemsByName = T{}
+			cc.sandbox.inventoryGear = T{}
+			cc.sandbox.gsGear = T{}
+			for k,v in pairs(gearswap.res.items) do
+				cc.sandbox.itemsBylongName[gearswap.res.items[k].enl:lower()] = k
+				cc.sandbox.itemsByName[gearswap.res.items[k].en:lower()] = k

```

Comment:
A note on using our resources. Please use `english` and `english_log` instead of `en` and `enl`, despite what it shows in the file. We use the short version in the file primarily to save on file size since the resources can get huge in some cases.

Also, I don't know how well that works with other GearSwap features, but consider using `name` and `name_log` instead of either. That will make it so that Japanese players can write their item names in Japanese and it will still match them, since `name` expands to `english` for English clients, and `japanese` for Japanese ones.